src/helpers.rs

```diff
@@ -50,6 +50,24 @@ pub(crate) fn clone_and_degit(url: &str, target: &Path) -> Result<()> {
 	Ok(())
 }
 
+/// Init a new git repo on creation of a parachain
+pub(crate) fn git_init(target: &Path, message: &str) -> Result<(), git2::Error> {
+	let repo = Repository::init(target)?;
+	let signature = repo.signature()?;
+
+	let mut index = repo.index()?;
+	index.add_all(["*"].iter(), IndexAddOption::DEFAULT, None)?;
+	let tree_id = index.write_tree()?;
+
+	let tree = repo.find_tree(tree_id)?;
+	let commit_id = repo.commit(Some("HEAD"), &signature, &signature, message, &tree, &[])?;
+
+	let commit_object = repo.find_object(commit_id, Some(git2::ObjectType::Commit))?;
```

Comment:
Can you briefly elaborate on what this piece of code achieves?

Reply:
- gets the index file of the repo and add all the files to it (staging area)
- creates a new git tree and write the index to it
- gets the tree from the `tree_id`
- creates a new commit, where, `signature` is the default author, commit message, the tree that was created and an empty vec of parent commit ids.
- resets the repo's working dir and index to the latest commit, to ensure repo's files match the state captured by the initial commit

Reply:
In the case of no git config for `name` and `email` does this approach work?

Reply:
if I'm not mistaken, git takes in the default user and system name if not specified explicitly. So yes, it should work, but I have to try and let you know.